src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.CoreCLR.cs

```diff
@@ -157,55 +158,55 @@ public override string ToString()
             return m_toString;
         }
 
-        public override int GetHashCode()
-        {
-            // See RuntimeMethodInfo.Equals() below.
-            if (IsGenericMethod)
-                return ValueType.GetHashCodeOfPtr(m_handle);
-            else
-                return base.GetHashCode();
-        }
+        public override int GetHashCode() => ValueType.GetHashCodeOfPtr(m_handle);
 
         public override bool Equals(object? obj)
         {
-            if (!IsGenericMethod)
-                return obj == (object)this;
+            if (IsGenericMethod)
```

Comment:
I think that this is way overengineered for what it does. You should be able to change it to just:
```
public override bool Equals(object? obj)
{
    // We cannot do simple object identity comparisons for generic methods.
    // Equals will be called in CerHashTable when RuntimeType+RuntimeTypeCache.GetGenericMethodInfo()
    // retrieve items from and insert items into s_methodInstantiations which is a 
    // CerHashtable.
    return obj is RuntimeMethodInfo m && m_handle == m.m_handle;
}
```